plugins/http/src/commands.rs

```diff
@@ -20,6 +20,64 @@ pub struct FetchResponse {
     url: String,
 }
 
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProxyConfig {
+    all: Option<String>,
+    http: Option<String>,
+    https: Option<String>,
+    basic_auth: Option<BasicAuth>,
+    no_proxy: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct BasicAuth {
+    username: String,
+    password: String,
+}
+
+fn attach_proxy(
+    settings: ProxyConfig,
+    builder: reqwest::ClientBuilder,
+) -> crate::Result<reqwest::ClientBuilder> {
+    let ProxyConfig {
+        all,
+        http,
+        https,
+        basic_auth,
+        no_proxy,
+    } = settings;
+
+    let mut builder = builder;
```

Comment:
this can be removed and `mut` can be added to the function argument
```rs
fn attach_proxy(
    settings: ProxyConfig,
    mut builder: reqwest::ClientBuilder,
) -> crate::Result<reqwest::ClientBuilder> {}
```